src/Model/MethodGo.cs

```diff
@@ -181,26 +157,59 @@ public string ResponderReturnSignature()
             {
                 return MethodReturnSig(ReturnType.Body.Cast<FutureTypeGo>().ResultTypeName);
             }
+            else if (IsPageable && !IsNextMethod)
+            {
+                return MethodReturnSig(ReturnType.Body.Cast<PageTypeGo>().ContentType.Name);
+            }
             return MethodReturnSignature();
         }
 
-        public string NextMethodName => $"{Name}NextResults";
+        /// <summary>
+        /// Returns the method return signature for the next results page method (e.g. "foo, bar").
+        /// </summary>
+        /// <returns>The method signature for the next results page method.</returns>
+        public string NextMethodReturnSignature()
+        {
+            return MethodReturnSig(ReturnType.Body.Cast<CompositeTypeGo>().UnwrapPageType().ContentType.Name);
+        }
+
+        /// <summary>
+        /// Returns the type name used as the parameter for the "next results" method.
+        /// </summary>
+        /// <returns>The "next results" method parameter type name.</returns>
+        public string LastResultsTypeName()
+        {
+            var type = ReturnValue().Body;
+            if (IsLongRunningOperation())
+            {
+                type = type.Cast<FutureTypeGo>().ResultType;
+            }
+            if (IsPageable && !IsNextMethod)
+            {
+                type = type.Cast<PageTypeGo>().ContentType;
+            }
+            return type.Name;
+        }
+
+        public string NextMethodName => $"{Name.ToCamelCase()}NextResults";
 
         public string PreparerMethodName => $"{Name}Preparer";
 
         public string SenderMethodName => $"{Name}Sender";
 
         public string ResponderMethodName => $"{Name}Responder";
 
+        public string ListAllMethodName => $"{Name}All";

```

Comment:
Unfortunately this naming convention is causing collisions, e.g. some swaggers define "List" and "ListAll" methods so this now creates a collision and an ugly "ListAllAll" method.  I am going to change this back to "Complete" but am open to suggestions.